scalameta/semantic/shared/src/main/protobuf/semanticdb.proto

```diff
@@ -1,25 +1,22 @@
 syntax = "proto3";
 
-package org.scalameta.semantic.v1.proto;
+package scala.meta.internal.semantic.proto;
 
-// scala.meta.semantic.v1.Database
 message Database {
-  // a database is treated as a sequence of database files to accommodate
-  // - incremental compilation, a database file maps one-to-one with a compilation unit.
-  // - simple merging of two databases, the bytes of two semanticdbs can in
-  //   essence be concatenated.
-  repeated DatabaseFile files = 1;
+  // A database is treated as a sequence of attributed source files.
+  // This accommodates incremental compilation, because database entries
+  // map one-to-one to compilation units.
+  // This also makes merging databases trivial, as the bytes of two databases
+  // can be simply concatenated to produce a new database.
+  repeated AttributedSource sources = 1;
 }
 
-message DatabaseFile {
-  Address address = 1;
-  repeated ResolvedName symbols = 2;
-  repeated CompilerMessage messages = 3;
-}
-
-message Address {
-  string path = 1; // Address.file
-  string contents = 2; // Address.snippet
+message AttributedSource {
+  reserved 1;
+  string path = 5;
```

Comment:
Following discussions from https://github.com/scalameta/scalameta/issues/777 .

I propose that we name this field `relative_path` and add a comment explaining that an absolute "source root" path is required at mirror construction time to turn `relative_path` into an absolute path.